src/main/java/work/lclpnet/ap2/game/treasure_hunter/TreasureHunterInstance.java

```diff
@@ -0,0 +1,169 @@
+package work.lclpnet.ap2.game.treasure_hunter;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.GameMode;
+import net.minecraft.world.World;
+import org.json.JSONArray;
+import work.lclpnet.ap2.api.base.Participants;
+import work.lclpnet.ap2.api.game.MiniGameHandle;
+import work.lclpnet.ap2.api.game.data.DataContainer;
+import work.lclpnet.ap2.impl.game.DefaultGameInstance;
+import work.lclpnet.ap2.impl.game.data.OrderedDataContainer;
+import work.lclpnet.ap2.impl.map.MapUtil;
+import work.lclpnet.kibu.access.entity.PlayerInventoryAccess;
+import work.lclpnet.kibu.hook.entity.PlayerInteractionHooks;
+import work.lclpnet.kibu.inv.item.ItemStackUtil;
+import work.lclpnet.kibu.plugin.hook.HookRegistrar;
+import work.lclpnet.lobby.game.impl.prot.ProtectionTypes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+public class TreasureHunterInstance extends DefaultGameInstance {
+    Random rand = new Random();
+    private final OrderedDataContainer data = new OrderedDataContainer();
+
+    public TreasureHunterInstance(MiniGameHandle gameHandle) {
+        super(gameHandle);
+
+        setDefaultGameMode(GameMode.SURVIVAL);
+    }
+
+    @Override
+    public void participantRemoved(ServerPlayerEntity player) {
```

Comment:
Implementation missing.
Needed when a participant quits.
If only one or less participants remain, the game should be ended.

Refer to MirrorHopInstance